Plugins/Source/KISAddonPickup.cs

```diff
@@ -873,10 +873,26 @@ public enum PickupMode {
 
   private IEnumerator WaitAndStopDrag() {
     yield return new WaitForFixedUpdate();
-    icon = null;
+    DisableIcon();
     draggedPart = null;
   }
 
+  // Sets icon, ensuring any old icon is Disposed
```

Comment:
Please, in the following changes stick with the code style (2 spaces indentation, no wrapping for opening bracket, etc.). I will change these ones myself once merge is done but it's better to not do second run on the code.